source/features/clean-conversation-headers.tsx

```diff
@@ -26,25 +30,31 @@ function initIssue(): void {
 
 function initPR(): void {
 	const observer = observe('.gh-header-meta .flex-auto:not(.rgh-clean-conversation-header)', {
-		add(byline) {
+		async add(byline) {
 			byline.classList.add('rgh-clean-conversation-header');
-			const isSameAuthor = select('.js-discussion > .TimelineItem:first-child .author')?.textContent === select('.author', byline)!.textContent;
-			const baseBranch = select('.commit-ref:not(.head-ref)', byline)!;
-			const isDefaultBranch = (baseBranch.firstElementChild as HTMLAnchorElement).pathname.split('/').length === 3;
 
-			// Removes: [octocat wants to merge 1] commit into github:master from octocat:feature
-			// Removes: [octocat] merged 1 commit into master from feature
-			for (const node of [...byline.childNodes].slice(isSameAuthor ? 0 : 2, pageDetect.isMergedPR() ? 2 : 4)) {
+			const author = select('.author', byline)!;
+			const isSameAuthor = pageDetect.isPRConversation() && author.textContent === select('.TimelineItem .author')!.textContent;
+
+			const base = select('.commit-ref', byline)!;
+			const baseBranch = base.title.split(':')[1];
+			const isDefaultBranch = baseBranch === await getDefaultBranch() || (pageDetect.isClosedPR() && baseBranch === 'master');
+
+			byline.childNodes[pageDetect.isClosedPR() ? (pageDetect.isMergedPR() ? 5 : 7) : 9].replaceWith(<> <ArrowLeftIcon/> </>);
+
+			// Removes: [octocat wants to merge 1 commit into] github:master from octocat:feature
+			// Removes: [octocat merged 1 commit into] master from feature
+			for (const node of [...byline.childNodes].slice(isSameAuthor ? 0 : 2, pageDetect.isMergedPR() ? 3 : 5)) {
```

Comment:
If not too noisy/long, maybe we can do the same instead of this loop. Alternatively as a simple cleanup option:
```suggestion
			const duplicateNodes = of [...byline.childNodes].slice(
			    isSameAuthor ? 0 : 2, 
			    pageDetect.isMergedPR() ? 3 : 5
			);
			for (const node of duplicateNodes) {
```

Reply:
Maybe use an array and push elements to delete into it?

Reply:
Is it shorter and/or more readable than that?

Reply:
You mean something like this? 

```suggestion
			const duplicateNodes = [];
			if (isSameAuthor) {
			    duplicateNodes.push(byline.childNodes[0]);
			    duplicateNodes.push(byline.childNodes[1]);
			);
			duplicateNodes.push(byline.childNodes[2]);
			duplicateNodes.push(byline.childNodes[3]);
			if (!pageDetect.isMergedPR()) {
			    duplicateNodes.push(byline.childNodes[4]);
			    duplicateNodes.push(byline.childNodes[5]);
			);
			for (const node of duplicateNodes) {
```

Obviously not, but I might have misunderstood